src/util/fields.py

```diff
@@ -756,8 +822,10 @@ def convert_field_to_property(
 
 
 # alias with a name related to the new API to declare property fields (company_dependent=True attribute)
-make_field_company_dependent = convert_field_to_property
-
+if version_gte("saas~17.5"):
+    make_field_company_dependent = convert_field_to_company_dependent
+else:
+    make_field_company_dependent = convert_field_to_property
```

Comment:
I don't like the fact that you have the two *public* methods available.

I propose this design instead.
```python

def _convert_field_to_property(
    cr, model, field, type, target_model=None, default_value=None, default_value_ref=None, company_field="company_id"
):
    # existing `convert_field_to_property` function
    ...
def _convert_field_to_company_dependent_jsonb(
     cr, model, field, type, target_model=None, default_value=None, default_value_ref=None, company_field="company_id"
):
     # new impl
     ...

def make_field_company_dependent(
    cr, model, field, type, target_model=None, default_value=None, default_value_ref=None, company_field="company_id"
):
    if table_exists(cr, "ir_property"):
        return _convert_field_to_property(...)
    else:
        return _convert_field_to_company_dependent_jsonb(...)

# retro-compatibility alias
convert_field_to_property = make_field_company_dependent
```